src/software/ai/hl/stp/play/halt_play/halt_play_fsm.cpp

```diff
@@ -0,0 +1,13 @@
+//
+// Created by grayson on 2024-10-19.
+//
+
+#include "software/ai/hl/stp/play/halt_play/halt_play_fsm.h"
+
+void HaltPlayFSM::updateStop(const Update& event)
+{
+    halt_tactics = {{std::make_shared<StopTactic>(), std::make_shared<StopTactic>(),
+                     std::make_shared<StopTactic>(), std::make_shared<StopTactic>(),
+                     std::make_shared<StopTactic>(), std::make_shared<StopTactic>()}};
```

Comment:
Consider using `std::generate` to initalize the vector and using this [constant](https://github.com/UBC-Thunderbots/Software/blob/master/src%2Fshared%2Fconstants.h#L151) for the vector size. You also don't need to reinitialize the vector and the tactics every time `updateStop` is called; you can just initialize the vector once in the constructor and reuse the same tactic instances

Reply:
So, to use std::generate, I would first have to initialize halt_tactics as vector of a vector with 8 values. Then I use std::generate to place the StopTactic inside each address in the innermost vector?

Not really sure how to implement this, and kind of struggling with initializing halt_tactics...

Reply:
Sorry I got mixed up, maybe `std::generate` isn't the best algorithm to use for this scenario. A better idea would be to create an empty vector and then use [`std::generate_n`](https://en.cppreference.com/w/cpp/algorithm/generate_n) to fill it with `n` newly constructed `StopTactic`s. You can use [`std::back_inserter`](https://en.cppreference.com/w/cpp/iterator/back_inserter) as the output iterator, which returns an iterator positioned at the end of the vector and calls `push_back` on the vector every time it's dereferenced. You can learn more about the [iterator pattern here](https://learn.microsoft.com/en-us/cpp/standard-library/iterators?view=msvc-170).

So your constructor would look something like:
```cpp
HaltPlayFSM::HaltPlayFSM() : halt_tactics()
{
    std::generate_n(std::back_inserter(halt_tactics), MAX_ROBOT_IDS_PER_SIDE, 
                    []() { return std::make_shared<StopTactic>(); });
}
```